flask/blueprints.py

```diff
@@ -66,7 +66,10 @@ def add_url_rule(self, rule, endpoint=None, view_func=None, **options):
         blueprint's name.
         """
         if self.url_prefix is not None:
-            rule = '/'.join((self.url_prefix, rule.lstrip('/')))
+            if rule:
+                rule = '/'.join((self.url_prefix, rule.lstrip('/')))
```

Comment:
I think this makes `@bp.route('/')` with prefix `/test` become `/test` instead of `/test/` now, which is also incorrect.

Reply:
see my test.

```
'/'.join(['/test', '']) => /tests/
```

Reply:
`prefix='/foo/', rule=''` results in `/foo` instead of what people would probably expect, `/foo/`. I think we need to take a different approach to merging, we're trying to cover too many corner cases.